elements/lisk-dpos/src/delegates_list.ts

```diff
@@ -328,67 +344,71 @@ export class DelegatesList {
 		_setVoteWeights(stateStore, voteWeights);
 	}
 
-	/**
-	 * Generate list of delegate public keys for the next round in database
-	 * WARNING: This function should only be called from `apply()` as we don't allow future rounds to be created
-	 */
-	public async createRoundDelegateList(
+	public async updateForgersList(
 		round: number,
+		randomSeed: ReadonlyArray<Buffer>,
 		stateStore: StateStore,
 	): Promise<void> {
-		debug(`Creating delegate list for round: ${round}`);
-		const forgersList = await getForgersList(stateStore);
-		const forgerListIndex = forgersList.findIndex(fl => fl.round === round);
-		// This gets the list before current block is executed
-		const delegateAccounts = await this.chain.dataAccess.getDelegateAccounts(
-			this.activeDelegates,
-		);
-		const updatedAccounts = stateStore.account.getUpdated();
-		// tslint:disable-next-line readonly-keyword
-		const updatedAccountsMap: { [address: string]: Account } = {};
-		// Convert updated accounts to map for better search
-		for (const account of updatedAccounts) {
-			updatedAccountsMap[account.address] = account;
+		// tslint:disable-next-line no-magic-numbers
+		if (randomSeed.length !== this.standbyDelegates) {
+			throw new Error(
+				`Exactly ${this.standbyDelegates} random seed must be provided`,
+			);
```

Comment:
That leaves the fact that there can be as many random seeds as many standby delegates we had. But the algorithm of calculating random seeds is restricted to two random seeds only, there is no definition of generating `n` number of random seeds.

Reply:
After discussing with research team, I remove the equality check of # of random seed and # of standby delegate
Now only checks the existence and uses modulo to loop through